src/graph/sampler.cc

```diff
@@ -865,4 +883,172 @@ DGL_REGISTER_GLOBAL("sampling._CAPI_LayerSampling")
     *rv = List<NodeFlow>(nflows);
   });
 
+void BuildCoo(const ImmutableGraph &g) {
```

Comment:
I think we need to put all these functions in anonymous namespace to avoid exporting these symbols.